eth/downloader/statesync.go

```diff
@@ -307,14 +310,14 @@ func (s *stateSync) loop() error {
 				s.d.dropPeer(req.peer.id)
 			}
 			// Process all the received blobs and check for stale delivery
-			if err := s.process(req); err != nil {
+			if err = s.process(req); err != nil {
 				log.Warn("Node data write error", "err", err)
-				return err
+				return
 			}
 			req.peer.SetNodeDataIdle(len(req.response))
 		}
 	}
-	return s.commit(true)
+	return
```

Comment:
Please don't use naked returns, it's very hard to see what's being returned.

Reply:
Also the previous logic called `s.commit(true)` when it reached this point, but you replaced it with `s.commit(false)` (in the defer), so the behavior changes.

Reply:
Sorry i can understand very well. I add the `s.commit(true)` in the defer because during the state node process procedure, some of the them can be invalid. If we return the error directly, we will lost the valid data committed before.